bundles/org.openhab.binding.irobot/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,260 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="irobot"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Sample Thing Type -->
+	<thing-type id="roomba">
+		<label>Roomba</label>
+		<description>A Roomba vacuum robot</description>
+
+		<channels>
+			<channel id="command" typeId="command"/>
+			<channel id="cycle" typeId="cycle"/>
+			<channel id="phase" typeId="phase"/>
+			<channel id="battery" typeId="battery"/>
+			<channel id="bin" typeId="bin"/>
+			<channel id="error" typeId="error"/>
+			<channel id="rssi" typeId="rssi"/>
+			<channel id="snr" typeId="snr"/>
+			<channel id="sched_mon" typeId="sched_switch">
+				<label>Schedule Mon</label>
+				<description>Monday schedule active</description>
+			</channel>
+			<channel id="sched_tue" typeId="sched_switch">
+				<label>Schedule Tue</label>
+				<description>Tuesday schedule active</description>
+			</channel>
+			<channel id="sched_wed" typeId="sched_switch">
+				<label>Schedule Wed</label>
+				<description>Wednesday schedule active</description>
+			</channel>
+			<channel id="sched_thu" typeId="sched_switch">
+				<label>Schedule Thu</label>
+				<description>Thirsday schedule active</description>
+			</channel>
+			<channel id="sched_fri" typeId="sched_switch">
+				<label>Schedule Fri</label>
+				<description>Friday schedule active</description>
+			</channel>
+			<channel id="sched_sat" typeId="sched_switch">
+				<label>Schedule Sat</label>
+				<description>Saturday schedule active</description>
+			</channel>
+			<channel id="sched_sun" typeId="sched_switch">
+				<label>Schedule Sun</label>
+				<description>Sunday schedule active</description>
+			</channel>
+			<channel id="schedule" typeId="schedule"/>
+			<channel id="edge_clean" typeId="edge_clean"/>
+			<channel id="always_finish" typeId="always_finish"/>
+			<channel id="power_boost" typeId="power_boost"/>
+			<channel id="clean_passes" typeId="clean_passes"/>
+		</channels>
+		<config-description>
+			<parameter name="ipaddress" type="text">
+				<label>IP Address</label>
```

Comment:
You could specify `<context>network-address</context>` to get a free validation.

Reply:
But what about arbitrary host name ? Like "my_roomba.livingroom.local" ? It's not forbidden to use /etc/hosts or local DNS. Or is it ?

Reply:
I've checked what Daikin binding does, it even calls the field "Hostname", but still uses this context. Apparently it works with hostnames too, done